parquet-thrift/src/main/java/org/apache/parquet/hadoop/thrift/ThriftReadSupport.java

```diff
@@ -41,21 +41,13 @@
 import org.apache.parquet.thrift.projection.FieldProjectionFilter;
 import org.apache.parquet.thrift.projection.StrictFieldProjectionFilter;
 import org.apache.parquet.thrift.projection.ThriftProjectionException;
-import org.apache.parquet.thrift.projection.deprecated.DeprecatedFieldProjectionFilter;
 import org.apache.parquet.thrift.struct.ThriftType.StructType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@Deprecated
 public class ThriftReadSupport<T> extends ReadSupport<T> {
   private static final Logger LOG = LoggerFactory.getLogger(ThriftReadSupport.class);
-
-  /**
-   * Deprecated. Use {@link #STRICT_THRIFT_COLUMN_FILTER_KEY}
-   * Accepts a ";" delimited list of globs in the syntax implemented by {@link DeprecatedFieldProjectionFilter}
-   */
-  @Deprecated
```

Comment:
Why not keeping them as-is?

Reply:
These are marked as deprecated in 2015. I think removing these is a first step before removing the whole class.

Reply:
part of my attempts to move up to later hadoop versions hit real problems here. and along with thrift in general -i had to grab a copy of an x86 MBP for my mac m1, but even there homebrew had decided to upgrade it, so it was a backup in a corner I ended up with. At which point I discovered that homebrew isn't a real package manager as you can't roll back...

Reply:
I'm also on a M1, and it takes quite a bit of effort to get Thrift compiled. I got a patch ready to update Thrift to the latest version. I'm not sure if we need to update `parquet-format` first, since the release is still on an older version. I'll give it a try in a moment.

Reply:
that or stick the old version into the repo...